routing/base/astar_algorithm.hpp

```diff
@@ -231,25 +229,17 @@ class AStarAlgorithm
       return bestDistance.at(queue.top().vertex);
     }
 
-    // p_f(v) = 0.5*(π_f(v) - π_r(v)) + 0.5*π_r(t)
-    // p_r(v) = 0.5*(π_r(v) - π_f(v)) + 0.5*π_f(s)
+    // p_f(v) = 0.5*(π_f(v) - π_r(v))
+    // p_r(v) = 0.5*(π_r(v) - π_f(v))
     // p_r(v) + p_f(v) = const. Note: this condition is called consistence.
     Weight ConsistentHeuristic(Vertex const & v) const
     {
       auto const piF = graph.HeuristicCostEstimate(v, finalVertex);
       auto const piR = graph.HeuristicCostEstimate(v, startVertex);
       if (forward)
-      {
-        /// @todo careful: with this "return" here and below in the Backward case
-        /// the heuristic becomes inconsistent but still seems to work.
-        /// return HeuristicCostEstimate(v, finalVertex);
-        return 0.5 * (piF - piR + m_piRT);
-      }
+        return 0.5 * (piF - piR);
       else
-      {
-        // return HeuristicCostEstimate(v, startVertex);
-        return 0.5 * (piR - piF + m_piFS);
-      }
+        return 0.5 * (piR - piF);
```

Comment:
The idea here was taken from the presentation at princeton.edu which is linked to in this file.
Since only the difference in its values matters, the heuristic function admits an arbitrary additive shift and it's more intuitive for it to be zero in the end of the route, so we choose a shift that does exactly this.